tests/infrastructure/virt-handler.go

```diff
@@ -47,36 +49,64 @@ var _ = DescribeInfra("virt-handler", func() {
 	var (
 		virtClient       kubecli.KubevirtClient
 		originalKubeVirt *v1.KubeVirt
-		nodesToEnableKSM []*k8sv1.Node
+		nodesToEnableKSM []string
 	)
 
-	type ksmTestFunc func() (*v1.KSMConfiguration, []*k8sv1.Node)
+	type ksmTestFunc func() (*v1.KSMConfiguration, []string)
 
-	getNodesWithKSMAvailable := func(virtCli kubecli.KubevirtClient) []*k8sv1.Node {
+	getNodesWithKSMAvailable := func(virtCli kubecli.KubevirtClient) []string {
 		nodes := libnode.GetAllSchedulableNodes(virtCli)
 
-		nodesWithKSM := make([]*k8sv1.Node, 0)
+		nodesWithKSM := make([]string, 0)
 		for _, node := range nodes.Items {
 			command := []string{"cat", "/sys/kernel/mm/ksm/run"}
 			_, err := tests.ExecuteCommandInVirtHandlerPod(node.Name, command)
 			if err == nil {
-				nodesWithKSM = append(nodesWithKSM, &node)
+				nodesWithKSM = append(nodesWithKSM, node.Name)
 			}
 		}
 		return nodesWithKSM
 	}
 
+	forceMemoryPressureOnNodes := func(nodes []string) {
+		for _, node := range nodes {
+			data := []byte(fmt.Sprintf(`{"metadata": { "annotations": {"%s": "%s", "%s": "%s"}}}`,
+				v1.KSMFreePercentOverride, "1.0",
+				v1.KSMPagesDecayOverride, "-300",
+			))
+			_, err := virtClient.CoreV1().Nodes().Patch(context.Background(), node, types.StrategicMergePatchType, data, metav1.PatchOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+
+	restoreNodes := func(nodes []string) {
+		for _, node := range nodes {
+			patchBytes := []byte(fmt.Sprintf(`[{"op": "remove", "path": "/metadata/annotations/%s"}, {"op": "remove", "path": "/metadata/annotations/%s"}]`,
+				strings.ReplaceAll(v1.KSMFreePercentOverride, "/", "~1"), strings.ReplaceAll(v1.KSMPagesDecayOverride, "/", "~1")))
+			_, err := virtClient.CoreV1().Nodes().Patch(context.Background(), node, types.JSONPatchType, patchBytes, metav1.PatchOptions{})
+			if err != nil {
+				node2, err2 := virtClient.CoreV1().Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err2).NotTo(HaveOccurred())
+				Expect(err).NotTo(HaveOccurred(), `patch:"%s" annotations:%#v`, string(patchBytes), node2.GetAnnotations())
+			}
+		}
+	}
+
 	BeforeEach(func() {
 		virtClient = kubevirt.Client()
 
 		nodesToEnableKSM = getNodesWithKSMAvailable(virtClient)
 		if len(nodesToEnableKSM) == 0 {
 			Fail("There isn't any node with KSM available")
 		}
+
+		forceMemoryPressureOnNodes(nodesToEnableKSM)
+
 		originalKubeVirt = util.GetCurrentKv(virtClient)
 	})
 
 	AfterEach(func() {
+		restoreNodes(nodesToEnableKSM)
```

Comment:
So sad global `CleanNodes` function does not clean the annotations

Reply:
Agreed